azurerm/internal/services/network/tests/network_interface_network_security_group_association_resource_test.go

```diff
@@ -0,0 +1,281 @@
+	return func(s *terraform.State) error {
+		client := acceptance.AzureProvider.Meta().(*clients.Client).Network.InterfacesClient
+		ctx := acceptance.AzureProvider.Meta().(*clients.Client).StopContext
+
+		// Ensure we have enough information in state to look up in API
+		rs, ok := s.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("Not found: %s", resourceName)
+		}
+
+		nicID, err := azure.ParseAzureResourceID(rs.Primary.Attributes["network_interface_id"])
+		if err != nil {
+			return err
+		}
+
+		nicName := nicID.Path["networkInterfaces"]
+		resourceGroup := nicID.ResourceGroup
+
+		read, err := client.Get(ctx, resourceGroup, nicName, "")
+		if err != nil {
+			return fmt.Errorf("Error retrieving Network Interface %q (Resource Group %q): %+v", nicName, resourceGroup, err)
+		}
+
+		read.InterfacePropertiesFormat.NetworkSecurityGroup = nil
+
+		future, err := azuresdkhacks.UpdateNetworkInterfaceAllowingRemovalOfNSG(ctx, client, resourceGroup, nicName, read)
+		if err != nil {
+			return fmt.Errorf("Error removing Network Security Group Association for Network Interface %q (Resource Group %q): %+v", nicName, resourceGroup, err)
+		}
+
+		if err = future.WaitForCompletionRef(ctx, client.Client); err != nil {
+			return fmt.Errorf("Error waiting for removal of Network Security Group Association for NIC %q (Resource Group %q): %+v", nicName, resourceGroup, err)
+		}
+
+		return nil
+	}
+}
+
+func testAccAzureRMNetworkInterfaceSecurityGroupAssociation_basic(data acceptance.TestData) string {
+	template := testAccAzureRMNetworkInterfaceSecurityGroupAssociation_template(data)
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_network_interface" "test" {
+  name                = "acctestni-%d"
+  location            = azurerm_resource_group.test.location
+  resource_group_name = azurerm_resource_group.test.name
+
+  ip_configuration {
+    name                          = "testconfiguration1"
+    subnet_id                     = azurerm_subnet.test.id
+    private_ip_address_allocation = "Dynamic"
+  }
+}
+
+resource "azurerm_network_interface_security_group_association" "test" {
+  network_interface_id      = azurerm_network_interface.test.id
+  network_security_group_id = azurerm_network_security_group.test.id
+}
+`, template, data.RandomInteger)
+}
+
+func testAccAzureRMNetworkInterfaceSecurityGroupAssociation_requiresImport(data acceptance.TestData) string {
+	template := testAccAzureRMNetworkInterfaceSecurityGroupAssociation_basic(data)
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_network_interface_security_group_association" "import" {
+  network_interface_id      = azurerm_network_interface_security_group_association.test.network_interface_id
+  network_security_group_id = azurerm_network_interface_security_group_association.test.network_security_group_id
+}
+`, template)
+}
+
+func testAccAzureRMNetworkInterfaceSecurityGroupAssociation_updateNIC(data acceptance.TestData) string {
```

Comment:
Looks like the NIC isn't in the template, so this will create and not update?

Reply:
that's intentional since this is defined in the `Basic` to be able to override it here